code/obj/item/gun/kinetic.dm

```diff
@@ -623,23 +623,67 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 	name = "Riot Shotgun"
 	desc = "A police-issue shotgun meant for suppressing riots."
 	icon = 'icons/obj/48x32.dmi'
-	icon_state = "shotty"
+	icon_state = "shotty-empty"
 	item_state = "shotty"
 	force = 15.0
 	contraband = 5
 	caliber = 0.72
 	max_ammo_capacity = 8
-	auto_eject = 1
+	auto_eject = 0
 	can_dual_wield = 0
 	two_handed = 1
 	has_empty_state = 1
 	gildable = 1
+	var/racked_slide = FALSE
+
 
 	New()
 		ammo = new/obj/item/ammo/bullets/abg
 		set_current_projectile(new/datum/projectile/bullet/abg)
 		..()
 
+	update_icon()
+		. = ..()
+		src.icon_state = "shotty" + (gilded ? "-golden" : "") + (racked_slide ? "" : "-empty" )
+
+	canshoot()
+		if (src.racked_slide)
+			return ..()
+		else
+			boutput(usr, "<span class='notice'>You need to rack the slide before you can fire!</span>") //Pain
+			return FALSE
```

Comment:
this is going to do weird things with stuff like gun components

Reply:
Also appears to triple-print the message

Reply:
The triple print is due to how many times canshoot() calls itself, and I didn't think it was a real problem.

**usr** was really the only way to do this without redoing guncode (that im aware of) and I think the negative of amusing gun component effects doesnt outweigh the upside of displaying this message

Reply:
fixed